[release] 1.0.0-beta.3 (#2616)
mui/base-ui: CHANGELOG.md

# Versions

## v1.0.0-beta.3

_Sep 3, 2025_

### General changes

- **Breaking change:** Base UI event details.
  Custom event callbacks provide BaseUIEventDetails object as their second parameter.
  This object contains the source event, reason and methods to customize the behavior (where applicable).

>   (#2382) by @atomiks

Reviewer: Would be good to show the change in signature from before, mainly for `onOpenChange`

### Alert Dialog

- **Breaking change:** Support `initialFocus` and `finalFocus` functions.
  The `initialFocus` and `finalFocus` props can be functions that return DOM elements to focus.

>   (#2536) by @atomiks

Reviewer: Should mention that the element must be returned directly (`() => ref.current`), rather than returning the ref itself

- Restore focus to popup when focused element is removed (#2479) by @atomiks
- Fix limitShift offset based on arrow size (#2571) by @atomiks

### Preview Card

- Fix `transform-origin` variable calculation when Positioner `sideOffset` is a function (#2511) by @atomiks
- Fix limitShift offset based on arrow size (#2571) by @atomiks

### Radio Group

- Return null in form data when no option selected (#2473) by @ingokpp

### Scroll Area

- Prevent pointer events from sibling portals triggering hover (#2542) by @KenanYusuf

### Select

- Fix stale `items` prop (#2397) by @atomiks
- Fix unexpected close when nested inside two popovers (#2481) by @atomiks
- Fix `onValueChange` type inference (#2372) by @atomiks
- Fix `transform-origin` variable calculation when Positioner `sideOffset` is a function (#2511) by @atomiks
- Reset state when selected item is removed (#2577) by @atomiks
- Fix `data-highlighted` and DOM focus item desync (#2569) by @atomiks
- Fix item click with `defaultOpen` prop (#2570) by @atomiks
- Fix scroll arrows not propagating scroll fully to start/end of list (#2523) by @atomiks
- Fix limitShift offset based on arrow size (#2571) by @atomiks

### Slider

> - **Breaking change:** The `input type="range"` element receives focus and the thumb div is no longer focusable, focus styles targeting the thumb.
>   For example `.Thumb:focus-visible` should use `.Thumb:has(:focus-visible)` instead.

Reviewer 2: ```suggestion
- **Breaking change:** Instead of the thumb div, the `input type="range"` element receives focus. Focus styles that were targeting the thumb, should be updated.
  For example `.Thumb:focus-visible` should be replaced with `.Thumb:has(:focus-visible)`.
```
